de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java

```diff
@@ -0,0 +1,243 @@
+import org.eclipse.e4.ui.services.EMenuService;
+import org.eclipse.e4.ui.workbench.modeling.ESelectionService;
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.widgets.Composite;
+
+import de.tu_bs.cs.isf.e4cf.core.file_structure.FileTreeElement;
+import de.tu_bs.cs.isf.e4cf.core.file_structure.WorkspaceFileSystem;
+import de.tu_bs.cs.isf.e4cf.core.file_structure.tree.sync.TreeSynchronization;
+import de.tu_bs.cs.isf.e4cf.core.gui.java_fx.util.FXMLLoader;
+import de.tu_bs.cs.isf.e4cf.core.stringtable.E4CEventTable;
+import de.tu_bs.cs.isf.e4cf.core.stringtable.E4CStringTable;
+import de.tu_bs.cs.isf.e4cf.core.util.RCPContentProvider;
+import de.tu_bs.cs.isf.e4cf.core.util.ServiceContainer;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.interfaces.IProjectExplorerExtension;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.listeners.ProjectExplorerKeyListener;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.stringtable.StringTable;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.view.ProjectExplorerView;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+import javafx.embed.swt.FXCanvas;
+import javafx.embed.swt.SWTFXUtils;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.TreeItem;
+import javafx.scene.image.ImageView;
+import javafx.scene.image.WritableImage;
+
+public class ProjectExplorerViewController {
+	private static final String PROJECT_EXPLORER_VIEW_FXML = "/ui/view/ProjectExplorerView.fxml";
+
+	private ProjectExplorerView view;
+
+	@Inject
+	ServiceContainer services;
+	
+	@Inject private ESelectionService _selectionService;
+	@Inject private IEventBroker _eventBroker;
+	@Inject private EMenuService _menuService;
+	
+	private ChangeListener<TreeItem<FileTreeElement>> changeListener;
+
+	private WorkspaceFileSystem workspaceFileSystem;
+	private Map<String, IProjectExplorerExtension> fileExtensions;
+
+	@PostConstruct
+	public void postConstruct(Composite parent, IEclipseContext context, WorkspaceFileSystem fileSystem)
+			throws IOException {
+		
+		FXCanvas canvas = new FXCanvas(parent, SWT.None);
+		FXMLLoader<ProjectExplorerView> loader = new FXMLLoader<ProjectExplorerView>(context, StringTable.BUNDLE_NAME,
+				PROJECT_EXPLORER_VIEW_FXML);
+		view = loader.getController();
+
+		getContributions();
+		
+		// Structure of Directories representing a Tree
+		FileTreeElement treeRoot = initializeInput(fileSystem);
+
+		TreeItem<FileTreeElement> root = buildTree(treeRoot, true, null);
+
+		view.projectTree.setRoot(root);
+		view.projectTree.setShowRoot(false);
+		
+		_menuService.registerContextMenu(canvas, StringTable.PROJECT_EXPLORER_CONTEXT_MENU_ID);
+
+		Scene scene = new Scene(loader.getNode());
+		canvas.setScene(scene);
+	
+		setupSelectionService();
+		
+		// listeners
+		scene.setOnKeyPressed(new ProjectExplorerKeyListener(context));

```

Comment:
// adding key listeners to the scene

Reply:
There also will be other listeners added here